static/js/src/cube/components/CubeHeader/Hero.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Col, Row } from "@canonical/react-components";
+import { Module } from "../../types";
+
+type Props = {
+  modules: Module[];
+  isLoading: boolean;
+  certifiedBadge: null | Record<string, string>;
+  hasEnrollments: boolean;
+};
+
+const Hero = ({
+  modules,
+  isLoading,
+  certifiedBadge,
+  hasEnrollments,
+}: Props) => {
+  const [heroStripIsShallow, setHeroStripIsShallow] = useState(false);
+  const [heroContentMarginTop, setHeroContentMarginTop] = useState("2rem");
+  const heroStrip = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        const width = Math.floor(entry.contentRect.width);
+        if (width < 1035) {
+          setHeroStripIsShallow(false);
+          setHeroContentMarginTop("2rem");
+        } else {
+          setHeroStripIsShallow(true);
+          setHeroContentMarginTop("-2rem");
+        }
+      });
+    });
+
+    if (heroStrip.current) {
+      resizeObserver.observe(heroStrip.current);
+    }
+  }, [heroStrip.current]);
+
+  const passedCourses = modules.reduce(
+    (passedCount, module) => passedCount + (module.status === "passed" ? 1 : 0),
+    0
+  );
+
+  const heading = isLoading
+    ? ""
+    : certifiedBadge
+    ? "CUBE 2020 complete"
+    : hasEnrollments
+    ? "Your road to CUBE"
+    : "The road to CUBE";
```

Comment:
I know I just asked you to use this syntax in the render function but don't think it scales well with nested conditions. Here I find it hard to parse what's going on

Reply:
I agree, fixed :+1: